### DISCUSSION TITLE
Exogenous Variables

### DISCUSSION THREAD

**Discussion by BrunoCavag** [Nixtla/mlforecast#98] [General]
Are you planning to add support for exogenous variables like in NeuralForecast? (static, lagged, and future exogenous)

**Comment by jmoralez** [comment]
Is [this](https://nixtla.github.io/mlforecast/forecast.html#dynamic-features) what you're looking for?

**Reply by BrunoCavag** [reply]
I was referring for covariates (both past and future), as the one available in darts

**Reply by jmoralez** [reply]
Can you be more specific? We currently support static and dynamic features, as described in the link above

**Reply by BrunoCavag** [reply]
For example, instead of using just the dynamic feature of the current timestamp, use multiple lags of the feature.

**Reply by jmoralez** [reply]
I see. At the moment you can do it by adding a column for each value you want to use as a feature and pass it to the dynamic_dfs argument of predict (these columns should also be present in the training set). I'll take a look at what the api looks like in darts

**Reply by elisevans** [reply]
Rather than creating the lags of these dynamic features myself, is there any way that MLForecast can handle it for me?

For example, say I have a boolean feature (column name 'is_xmas') which tells me whether it is Christmas day on a datestamp, and I want to add another feature which is 'is_xmas' lagged by 1. Rather than adding columns 'is_xmas_lag1' myself to my df, and then passing the df to .predict or .cross_validate with dynamic_features = ['is_xmas', 'is_xmas_lag1'] specified, is there a way to specify a set of dynamic_features that MLForecast should lag, and exactly by how much?

I'm thinking something like dynamic_features = ['is_xmas'], dynamic_lags = [1], or perhaps dynamic_features = ['is_xmas', 'is_easter', 'is_halloween'], dynamic_lags = [1, 3, None], or similar. Hopefully this is clear.
